core/org.wso2.carbon.pax-logging-log4j2-plugins/src/main/java/org/wso2/carbon/log4j2/plugins/MemoryAppender.java

```diff
@@ -0,0 +1,145 @@
+     */
+    private MemoryAppender(final String name, final Filter filter,
+            final Layout<? extends Serializable> layout, final boolean ignoreExceptions) {
+        super(name, filter, layout, ignoreExceptions);
+    }
+
+    /**
+     * Creates a CarbonMemoryAppender instance with
+     * attributes configured in log4j2.properties.
+     *
+     * @param name appender name
+     * @param filter null if not specified
+     * @param layout pattern of log messages
+     * @param ignore default is true
+     * @return intance of MemoryAppender
+     */
+    @PluginFactory
+    public static MemoryAppender createAppender(@PluginAttribute("name") final String name,
+                                                @PluginElement("Filters") final Filter filter,
+                                                @PluginElement("Layout") Layout<? extends Serializable> layout,
+                                                @PluginAttribute("ignoreExceptions") final String ignore) {
+        if (name == null) {
+            LOGGER.error("No name provided for MemoryAppender");
+            return null;
+        } else {
+            if (layout == null) {
+                layout = PatternLayout.createDefaultLayout();
+            }
+            final boolean ignoreExceptions = Booleans.parseBoolean(ignore, true);
+            return new MemoryAppender(name, filter, layout, ignoreExceptions);
+        }
+    }
+
+    /**
+     * This is the overridden method from the Appender interface.
+     * This allows to write log events to preferred destination.
+     *
+     * Converts the default log events to tenant aware log events and writes to a CircularBuffer
+     *
+     * @param logEvent the LogEvent object
+     */
+    @Override
+    public void append(LogEvent logEvent) {
+        if (circularBuffer != null) {
+            MutableLogEvent mutableLogEvent = new MutableLogEvent(logEvent);
+            circularBuffer.append(mutableLogEvent.getMutableLogEvent());
+        }
+    }
+
+    /**
+     * Taken from the previous MemoryAppender
+     */
+    public CircularBuffer<LogEvent> getCircularQueue() {
+        return circularBuffer;
+    }
+
+    /**
+     * Taken from the previous MemoryAppender
+     */
+    public void setCircularBuffer(CircularBuffer<LogEvent> circularBuffer) {
+        this.circularBuffer = circularBuffer;
+    }
+
+    /**
+     * Taken from the previous MemoryAppender
+     */
+    public void activateOptions() {
+        if (bufferSize < 0) {
+            if (circularBuffer == null) {
+                this.circularBuffer = new CircularBuffer<LogEvent>();
+            }
+        } else {
+            this.circularBuffer = new CircularBuffer<LogEvent>(bufferSize);
+        }
+    }
+
+    /**
+     * Taken from the previous MemoryAppender
+     */
+    public int getBufferSize() {
+        return bufferSize;
+    }
+
+    /**
+     * Taken from the previous MemoryAppender
```

Comment:
Improve the doc comment, explain what this method does, arguments and return type

Reply:
Fixed in 8f514c9